pkgs/tools/filesystems/dwarfs/default.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv
+, bison
+, boost
+, cmake
+, double-conversion
+, fetchFromGitHub
+, fmt_8
+, fuse3
+, gflags
+, git
+, glog
+, jemalloc
+, lib
+, libarchive
+, libdwarf
+, libevent
+, libunwind
+, lz4
+, openssl
+, pkgconfig
+, ronn
+, xxHash
+, xz
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwarfs";
+  version = "0.6.1";
+  src = fetchFromGitHub {
+    owner = "mhx";
+    repo = "dwarfs";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-bGJkgcq8JxueRTX08QpJv1A0O5wXbiIgUY7BrY0Ln/M=";
+  };
+
+  # Needs tagged git repo for generating some files
+  # leaveDotGit alone isn't enough and breaks reproducible hashing
+  preConfigure = ''
+    git init
+    git add .
+    git -c user.name=root -c user.email=root@localhost commit -m "v${version}" -q
+    git -c user.name=root -c user.email=root@localhost tag "v${version}" -m "v${version}"
+  '';
+
+  # ...but the files are still kinda ugly, so let's make the data prettier for wherever it gets used ;)
+  postConfigure = ''
+    sed -i \
+      -e 's/\(PRJ_GIT_REV = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_DESC = \).*/\1"${version}";/' \
+      -e 's/\(PRJ_GIT_BRANCH = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_ID = \).*/\1"${version} from Nixpkgs :)";/' \
+      ../src/dwarfs/version.cpp
+  '';
+
+  cmakeFlags = [
+    "-DPREFER_SYSTEM_ZSTD=ON"
+    "-DPREFER_SYSTEM_XXHASH=ON"
+  ];
+
+  nativeBuildInputs = [
+    bison
+    boost.dev
+    cmake
+    fmt_8
+    git
+    libdwarf.dev
+    libevent.dev
+    pkgconfig
+    ronn
+  ];
+
+  buildInputs = [
+    boost
+    double-conversion
+    fmt_8
+    fuse3
+    gflags
+    glog
+    jemalloc
+    libarchive
+    libunwind
+    lz4
+    openssl
+    xxHash
+    xz
+    zstd
+ ];
+
+  meta = with lib; {
+    description = "A fast high compression read-only file system";
+    homepage = "https://github.com/mhx/dwarfs";
+    license = licenses.gpl3Only;
```

Comment:
All the source files I checked include the text `or (at your option) any later version` which would suggest this is `gpl3Plus`. Did you see something different?

Reply:
I actually hadn't looked at the headers in any of the source files, only having looked at the license. Thanks for catching that.